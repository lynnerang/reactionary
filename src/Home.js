import React from 'react';
import './Home.scss';
import icon from './images/flashicon.png';
import { NavLink } from 'react-router-dom';


function Home (props) {
    return (
      <>
        <h2>Home</h2>
        <div className='instructions'>
          <p>Welcome to Reactionary, the flash card app for studying up on your React terminology!  Here you can:</p>
        </div>
        <div className='home-layout'>
          <div className='home-options'>
            <div className='home-option'>
              <NavLink className='home-link' to='/play'>
                <h3 className='home-option-heading'>Play a Game!</h3>
                <p>Choose from multiple game styles, like entering a guess or choosing a guess from multiple choices.</p>
              </NavLink>
            </div>
            <div className='home-option'>
              <NavLink className='home-link' to='/mycards'>
                <h3 className='home-option-heading'>Make Your Own Flash Cards!</h3>
                <p>Create your very own flash cards that will be included in the Review and Guess Mode games.</p>
              </NavLink>
            </div>
            <div className='home-option'>
              <NavLink className='home-link' to='/stats'>
                <h3 className='home-option-heading'>View Your Stats!</h3>
                <p>See how many games you've played, what your overall top score is, and more.</p>
              </NavLink>
            </div>
            <div className='home-option'>
              <NavLink className='home-link' to='/links'>
                <h3 className='home-option-heading'>See Helpful React Resources!</h3>
                <p>Get links to a variety of great resources for brushing up on your React skills and knowledge.</p>
              </NavLink>
            </div>
          </div>
          <div className='home-image-area'>
            <img className='home-image' src={icon} alt='flashcards icon' />
          </div>
        </div>
      </>
    )
}

export default Home;










