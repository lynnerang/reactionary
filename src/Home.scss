@import './variables.scss';


.home-layout {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.home-options {
  flex: 2;
}

.home-option {
  max-width: 600px;
  border: 1px solid rgba(131, 131, 131, 0.475);
  border-radius: 5px;
  margin: 15px 0px;
  padding: 15px;
  cursor: pointer;

  p {
    font-size: 18px;
  }

  &:hover {
    h3 {
      color: $blue;
    }
    border-color: $blue;
  }
}

.home-link {
  text-decoration: none;
}

.home-option-heading {
  font-size: 20px;
}

.home-image-area {
  flex: 1.2;
}

.home-image {
  height: 400px;
  opacity: .6;
}