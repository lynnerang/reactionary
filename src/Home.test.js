import React from 'react';
import Home from './Home.js';
import { shallow } from 'enzyme';


describe('Home', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <Home 
      />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});