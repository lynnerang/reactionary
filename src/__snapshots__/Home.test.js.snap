// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Should match the snapshot 1`] = `
<Fragment>
  <h2>
    Home
  </h2>
  <div
    className="instructions"
  >
    <p>
      Welcome to Reactionary, the flash card app for studying up on your React terminology!  Here you can:
    </p>
  </div>
  <div
    className="home-layout"
  >
    <div
      className="home-options"
    >
      <div
        className="home-option"
      >
        <NavLink
          className="home-link"
          to="/play"
        >
          <h3
            className="home-option-heading"
          >
            Play a Game!
          </h3>
          <p>
            Choose from multiple game styles, like entering a guess or choosing a guess from multiple choices.
          </p>
        </NavLink>
      </div>
      <div
        className="home-option"
      >
        <NavLink
          className="home-link"
          to="/mycards"
        >
          <h3
            className="home-option-heading"
          >
            Make Your Own Flash Cards!
          </h3>
          <p>
            Create your very own flash cards that will be included in the Review and Guess Mode games.
          </p>
        </NavLink>
      </div>
      <div
        className="home-option"
      >
        <NavLink
          className="home-link"
          to="/stats"
        >
          <h3
            className="home-option-heading"
          >
            View Your Stats!
          </h3>
          <p>
            See how many games you've played, what your overall top score is, and more.
          </p>
        </NavLink>
      </div>
      <div
        className="home-option"
      >
        <NavLink
          className="home-link"
          to="/links"
        >
          <h3
            className="home-option-heading"
          >
            See Helpful React Resources!
          </h3>
          <p>
            Get links to a variety of great resources for brushing up on your React skills and knowledge.
          </p>
        </NavLink>
      </div>
    </div>
    <div
      className="home-image-area"
    >
      <img
        alt="flashcards icon"
        className="home-image"
        src="head2.png"
      />
    </div>
  </div>
</Fragment>
`;
